src/Courses/Settings/OverridesForm.tsx

```diff
@@ -0,0 +1,238 @@
+            }
+        })();
+    }, [topic, userId]);
+
+    // Get Question Override information
+    useEffect(()=>{
+        if (!problem) return;
+
+        (async () => {
+            try {
+                const res = await getQuestion({id: problem.id, userId});
+                const questionData = res.data.data;
+                console.log(questionData);
+                // If there are overrides for the selected user, overwrite the default dates in the object.
+                // Right now, we expect only one override per topic to be returned.
+                if (questionData.studentTopicQuestionOverride?.length === 1) {
+                    _.assign(questionData, questionData.studentTopicQuestionOverride[0]);
+                }
+
+                setValue('maxAttempts', questionData.maxAttempts);
+            } catch (e) {
+                console.error(`Question ${problem.id} or User ${userId} does not exist!`, e);
+            } finally {
+                setFormLoading(false);
+            }
+        })();
+
+    }, [problem, userId]);
+
+    const updateTopic = async (courseTopicContentId: number, userId: number, extensions: {startDate: Moment, endDate: Moment, deadDate: Moment}) => {
+        try {
+            const res = await extendTopic({courseTopicContentId, userId, extensions});
+            console.log(res);
+        } catch (e) {
+            console.error(e);
+            throw e;
+        }
+    };
+
+    const updateQuestions = async (courseTopicQuestionId: number, userId: number, extensions: {maxAttempts: number}) => {
+        try {
+            const res = await extendQuestion({courseTopicQuestionId, userId, extensions});
+            console.log(res);
+        } catch (e) {
+            console.error(e);
+            throw e;
+        }
+    };
+
+    const onSubmit = async (extensions: {startDate: Moment, endDate: Moment, deadDate: Moment} | {maxAttempts: number}) => {
+        setSubmitLoading(true);
+        console.log(extensions);
+        if (problem) {
+            await updateQuestions(problem.id, userId, extensions as {maxAttempts: number});
+        } else if (topic) {
+            await updateTopic(topic.id, userId, extensions as {startDate: Moment, endDate: Moment, deadDate: Moment});
+        } else {
+            console.error('Unhandled override case.');
+        }
+        setSubmitLoading(false);
+    };
+
+    const renderQuestionOverrideForm = (question: ProblemObject) => (
+        <Grid item container md={12} alignItems='flex-start' justify="center">
+            <Grid item md={4}>
+                <TextField inputRef={register} name="maxAttempts" defaultValue={question.maxAttempts} label='Max Attempts' />
+            </Grid>
+        </Grid>
+    );
+
+    const renderTopicOverrideForm = (topic: NewCourseTopicObj) => (
+        <>
+            <Grid item md={12}>
+                <MuiPickersUtilsProvider utils={MomentUtils}>
+                    <Controller
```

Comment:
We've had issues with doing this with the `Controller`, is there any advantage
On the topic modal you couldn't control `Controller` with the value state
Unless there's a reason to we should keep it consistent

Reply:
Talked about it in scrum and @PanopticaRising showed me that it works as expected, might have been how we did the `as` in the past